PokeAlarm/Alarms/Discord/DiscordAlarm.py

```diff
@@ -253,11 +253,14 @@ def send_alert(self, alert, info):
                     "title": replace(alert["title"], info),
                     "url": replace(alert["url"], info),
                     "description": replace(alert["body"], info),
-                    "thumbnail": {"url": replace(alert["icon_url"], info)},
                     "fields": self.replace_fields(alert["fields"], info),
                 }
             ]
 
+            thumbnail_url = replace(alert["icon_url"], info)
+            if thumbnail_url != "":
```

Comment:
Shouldn't we be validating whether `thumbnail_url` is not `null` or `undefined` here as well as a fallback in case the call to `replace()` returns an unexpected result?

Reply:
The replace function in Alarm.replace can't return something other than a string.